platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -130,106 +139,103 @@ static void mbed_minimal_putchar(char *buffer, size_t length, int *result, char
 }
 
 /**
- * @brief      Print signed integer.
+ * @brief      Print integer in signed, unsigned or hexadecimal format.
  *
- * @param      buffer  The buffer to store output (NULL for stdout).
- * @param[in]  length  The length of the buffer.
- * @param      result  The current output location.
- * @param[in]  value   The value to be printed.
+ * @param      buffer           The buffer to store output (NULL for stdout).
+ * @param[in]  length           The length of the buffer.
+ * @param      result           The current output location.
+ * @param[in]  value            The value to be printed.
+ * @param      type             The type of integer format that shall be printed (signed, unsigend or hexadecimal)
+ * @param      width_size       The width modifier.
+ * @param      prepend_zeros    Flag to prepends zeros when the width_size is greater than 0
  */
-static void mbed_minimal_formatted_string_signed(char *buffer, size_t length, int *result, MBED_SIGNED_STORAGE value, FILE *stream)
+static void mbed_minimal_formatted_string_integer(
+    char *buffer,
+    size_t length,
+    int *result,
+    MBED_UNSIGNED_STORAGE value,
+    integer_type_t type,
+    int width_size,
+    bool prepend_zeros,
+    FILE *stream)
 {
-    MBED_UNSIGNED_STORAGE new_value = 0;
+    /* allocate 3 digits per byte */
+    char scratch[sizeof(MBED_UNSIGNED_STORAGE) * 3] = { 0 };
 
-    /* if value is negative print sign and treat as positive number */
-    if (value < 0) {
-        /* write sign */
-        mbed_minimal_putchar(buffer, length, result, '-', stream);
+    int index = 0;
 
-        /* get absolute value using two's complement */
-        new_value = ~((MBED_UNSIGNED_STORAGE) value) + 1;
-    } else {
-        new_value = value;
-    }
-
-    /* use unsigned long int function */
-    mbed_minimal_formatted_string_unsigned(buffer, length, result, new_value, stream);
-}
+    const char filler = prepend_zeros ? '0' : ' ';
+    memset(scratch, filler, sizeof(scratch));
 
-/**
- * @brief      Print unsigned integer.
- *
- * @param      buffer  The buffer to store output (NULL for stdout).
- * @param[in]  length  The length of the buffer.
- * @param      result  The current output location.
- * @param[in]  value   The value to be printed.
- */
-static void mbed_minimal_formatted_string_unsigned(char *buffer, size_t length, int *result, MBED_UNSIGNED_STORAGE value, FILE *stream)
-{
-    /* treat 0 as a corner case */
-    if (value == 0) {
-        mbed_minimal_putchar(buffer, length, result, '0', stream);
-    } else {
-        /* allocate 3 digits per byte */
-        char scratch[sizeof(MBED_UNSIGNED_STORAGE) * 3] = { 0 };
+    bool negativ_value = false;
 
-        size_t index = 0;
+    if (type == INT_SIGNED) {
+        MBED_SIGNED_STORAGE integer = value;
+        if (integer < 0) {
+            /* get absolute value using two's complement */
+            value = ~((MBED_UNSIGNED_STORAGE) integer) + 1;
```

Comment:
I don't think the temporary variable is needed or easy to read, as it is casted back-and-forth.
```suggestion
        if ((MBED_SIGNED_STORAGE) value < 0) {
            /* get absolute value using two's complement */
            value = ~value + 1;
```